deployment/ccip/changeset/deploy.go

```diff
@@ -261,10 +264,28 @@ func DeployPrerequisiteContracts(e deployment.Environment, ab deployment.Address
 			return err
 		}
 		e.Logger.Infow("deployed router", "addr", routerContract.Address)
-		r = routerContract.Contract
 	} else {
 		e.Logger.Infow("router already deployed", "addr", chainState.Router.Address)
 	}
+	if mc3 == nil {
```

Comment:
TODO need to update this based on deployOpts here https://github.com/smartcontractkit/chainlink/pull/15349/files